js/modules/k6/grpc/client.go

```diff
@@ -459,8 +459,17 @@ func (c *Client) parseInvokeParams(paramsVal goja.Value) (*invokeParams, error)
 			}
 			for hk, kv := range rawHeaders {
 				// TODO(rogchap): Should we manage a string slice?
-				strval, ok := kv.(string)
-				if !ok {
+				// The spec defines that Binary-valued keys end in -bin
+				//  https://grpc.io/docs/what-is-grpc/core-concepts/#metadata
+				var strval string
+				if strings.HasSuffix(hk, "-bin") {
+					var binval []byte
+					binval, ok = kv.([]byte)
+					if !ok {
+						return result, fmt.Errorf("metadata %q value must be binary", hk)
+					}
+					strval = string(binval)
```

Comment:
Can we link to https://github.com/grpc/grpc-go/blob/master/Documentation/grpc-metadata.md#storing-binary-data-in-metadata so that whoever comes after us knows why this is how it is done instead of keeping it in binary and giving it to the grpc library as binary